Subject: Patch to insure that crash_kdump is called when soft-reset is used
From: wilder@us.ibm.com
References: 159713 - LTC22581

unless panic_on_oops is set it is possible for none of the cpus to become the
crashing cpu when initiating a dump with a soft-reset.  Thus the boot of the
kdump kernel is never started.  I have corrected this issue by explicitly
checking if a soft-reset has been initiated. 

Signed-off-by: Olaf Hering <olh@suse.de>
---
 arch/powerpc/kernel/traps.c |    3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

Index: linux-2.6.16/arch/powerpc/kernel/traps.c
===================================================================
--- linux-2.6.16.orig/arch/powerpc/kernel/traps.c
+++ linux-2.6.16/arch/powerpc/kernel/traps.c
@@ -164,7 +164,8 @@ int die(const char *str, struct pt_regs 
 	bust_spinlocks(0);
 	spin_unlock_irq(&die_lock);
 
-	if (kexec_should_crash(current))
+	if (kexec_should_crash(current) ||
+		cpu_isset(smp_processor_id(),cpus_in_sr) )
 		crash_kexec(regs);
 	crash_kexec_secondary(regs);
 
